wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java

```diff
@@ -166,8 +167,13 @@ protected boolean doHandleRequest(final Request request) {
 	 * Reset validation state.
 	 */
 	private void resetValidationState() {
-		getOrCreateComponentModel().validFileType = true;
-		getOrCreateComponentModel().validFileSize = true;
+		// Update the model only if the original state has changed
+		if (!getComponentModel().validFileSize) {
+		    getOrCreateComponentModel().validFileType = true;
```

Comment:
Why are you testing `validFileSize` before setting `validFileType`? Are you ure this is correct?